### DISCUSSION TITLE
Preview Relase 2.0.0

### DISCUSSION THREAD

**Discussion by UnseenWizzard** [Dynatrace/dynatrace-configuration-as-code#813] [General]
### :warning: Preview Release

This is a pre-release of monaco 2.0.0.

- this does not yet reflect the full content or quality of the final release
- no official support is provided for the pre-release version
- please let us know your feedback https://github.com/Dynatrace/dynatrace-configuration-as-code/discussions/806

### Monaco 2.0.0 Changes

Version 2.x tries to achieve three main goals: 

- Make the CLI and configuration clearer to understand​
  - With more explicit commands​
  - Structured configuration YAML​
- Decouple from the file-path 
  - Monaco 1.x was impacted by the location it was executed in, in relation to the configuration​
- Provide general support for Settings 2.0 configuration​
  - Monaco 2.0 does not need to be updated to support future settings schemas​
  - In part possible due to the new structure​
  - :warning: In this preview release, re-applying downloaded settings objects to the same environment may result in duplicated settings. Manual cleanup is needed for this case. Creating new settings objects works without limitations.

### Getting Started

A preview of the documentation for 2.0.0 is available [here](https://www.dynatrace.com/support/help/shortlink/configuration-as-code)

Also, check out the [Observability Clinic introducing Monaco 2.0](https://dt-url.net/monaco-observability-clinic).

And the [related sample configurations](https://github.com/Dynatrace/dynatrace-configuration-as-code-samples/tree/main/observability_clinic_sample)

<hr /><em>This discussion was created from the release <a href='https://github.com/Dynatrace/dynatrace-configuration-as-code/releases/tag/v2.0.0-rc.4'>Preview Relase 2.0.0</a>.</em>

**Comment by ibarkowski** [comment]
Hi, good work! I'm during PoC with Monaco2.0, and I want to use Settings2.0 standard API from the begining instead of using old apis. This version do not support update of objects that already exists in environment. Do you have any ETA for this?

**Reply by UnseenWizzard** [reply]
Hi @ibarkowski, this is limited by the Settings framework itself and will be addressed in upcoming Dynatrace releases. 
From Dynatrace version 1.261 it will be possible to update most Settings with some limitations, from version 1.262 fully.

The final monaco 2.0 release will check the Dynatrace version it's running against and inform you if a Setting could not be updated. 
(The pre-release is still missing this handling).

**Reply by UnseenWizzard** [reply]
FYI a similar question was raised in the Community, and I gave a full answer and some insights into how Monaco identifies Settings and why this will need Dynatrace 1.262 to be fully functional there: 
https://community.dynatrace.com/t5/Dynatrace-API/Settings-API-how-can-externalId-be-updated-on-an-object/m-p/205492

**Comment by agardnerIT** [comment]
Looking forward to experimenting with this :)

Will Monaco 2.0 have a hooks concept as part of the lifecycle of monaco running? For example, send a [cloudevent](https://cloudevents.io) a notification when I finish a `config apply`?

Perhaps `before`, `after`, `error` and `finally`?

**Reply by UnseenWizzard** [reply]
That is not something we currently plan for. 
Please use the [Dynatrace Community](https://community.dynatrace.com/) to submit that as a product idea! :)
